### PR TITLE
Prevent location updates to crash on stopping VisionManager

### PR DESCRIPTION
**Pull request by chezzdev**
Checks:

* [x] Update changelog
* [x] Rebase to `dev` branch
* [x] Assign reviewers

Linked issues:
Resolves https://github.com/mapbox/mapbox-vision/issues/1962

### CODE REVIEW

File: MapboxVision/Services/LocationManager.swift [hunk 1]
```diff
@@ -41,11 +41,12 @@ final class LocationManager: NSObject, CLLocationManagerDelegate {
     }
 
     func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        guard let handler = locationHandler else { return }
+        guard let handler = locationHandler, isStarted else { return }
```
Comment (dersim-davaod) [inline MapboxVision/Services/LocationManager.swift]: Should we set `locationManager.locationHandler` to `nil` in `stop()` method in `RealtimeDataProvider` class in `DataProvider.swift`?

And correspondingly, move initialization for handlers from `init(dependencies: Dependencies)` to `start()`?

Comment (chezzdev) [reply MapboxVision/Services/LocationManager.swift]: I'm ok with that. Added.

Comment (dersim-davaod) [reply MapboxVision/Services/LocationManager.swift]: Is it enough to check for `isStarted` only now?

File: MapboxVision/Services/LocationManager.swift [hunk 2]
```diff
@@ -41,11 +41,12 @@ final class LocationManager: NSObject, CLLocationManagerDelegate {
...
         locations.forEach(handler)
     }
 
     func locationManager(_ manager: CLLocationManager, didUpdateHeading newHeading: CLHeading) {
+        guard isStarted else { return }
```
Comment (dersim-davaod) [inline MapboxVision/Services/LocationManager.swift]: We do not check the `headingHandler` for being `nil` here as for `locationManager:didUpdateLocations:` method - might be a bit confusing IMHO.

Or we can leave only one check for `isStarted` above (please take a look at related comment above).

Comment (chezzdev) [reply MapboxVision/Services/LocationManager.swift]: The guard in `didUpdateLocation` is there to unwrap handler before passing it to foreach. There's no need for it here.

Comment (dersim-davaod) [reply MapboxVision/Services/LocationManager.swift]: Thanks for the explanation!

**Pull request merged**